src/Microsoft.OData.Core/UriParser/SyntacticAst/CountSegmentToken.cs

```diff
@@ -0,0 +1,122 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="CountSegmentToken.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if ODATA_CLIENT
+namespace Microsoft.OData.Client.ALinq.UriParser
+#else
+namespace Microsoft.OData.UriParser
+#endif
+{
+    /// <summary>
+    /// Lexical token representing the $count segment in a path.
+    /// </summary>
+    public sealed class CountSegmentToken : PathToken
+    {
+        /// <summary>
+        /// The name of this token, which in this case is always "$count".
+        /// </summary>
+        private readonly string identifier;
+
+        /// <summary>
+        /// The instance to count on.
+        /// </summary>
+        private QueryToken nextToken;
+
+        /// <summary>
+        /// The token representing $filter.
+        /// </summary>
+        private QueryToken filterOption;
+
+        /// <summary>
+        /// The token representing $search.
+        /// </summary>
+        private QueryToken searchOption;
+
+        /// <summary>
+        /// Create a CountSegmentToken given the Identifier and the NextToken.
+        /// </summary>
+        /// <param name="identifier">The name of this token, which in this case is always "$count".</param>
+        /// <param name="nextToken">The instance to count on.</param>
+        public CountSegmentToken(string identifier, QueryToken nextToken)

```

Comment:
>string identifier,  [](http://example.com/codeflow?start=33&length=19)

if this is always $count, why bother passing it?  Why not just return QueryOptionCount on line 79?

Reply:
Fixed